src/DataPersister/CommandRequestDataPersister.php

```diff
@@ -28,21 +35,34 @@ public function supports($data): bool
     /**
      * {@inheritdoc}
      */
-    public function persist($data)
+    public function persist($data): object
     {
-        $this->handleCommand($data);
+        if (!$data instanceof CommandRequestInterface) {
+            throw new \LogicException(sprintf('Expected data to be "%s".', CommandRequestInterface::class));
+        }
+
+        $this->commandBus->dispatch($data->getCommand());
+
+        if ($data instanceof QueryRequestInterface) {
+            $data = $this->handle($data->getQuery());
```

Comment:
The query is some kind of command, is that correct? Why do we handle commands and queries differently?

Reply:
> CQRS stands for Command Query Responsibility Segregation

It's the architecture in ShopApiPlugin, as far as I could tell. But the "query" part was done using some weird "view repositories" before this.